persistence/persistence-mybatis/src/test/java/org/cbioportal/persistence/mybatis/SampleMyBatisRepositoryTest.java

```diff
@@ -340,7 +340,7 @@ public void getSamplesByInternalIds() throws Exception {
      
     @Test
     public void getSamplesByKeyword() {
-         List<Sample> result = sampleMyBatisRepository.getAllSamples("TCGA-A1-A0SB", "SUMMARY", 10, 0, null, null);
+         List<Sample> result = sampleMyBatisRepository.getAllSamples("TCGA-A1-A0SB", null, "SUMMARY", 10, 0, null, null);
```

Comment:
here, the unit test is being run in an unauthenticated mode I guess ... where we can pass "null" as the list of available studies and we still get back the expected response ... because all studies in the test database are considered public I guess. This means that this unit test is not testing the study-filtering logic in the SampleMapper. I am assuming the proper behavior was observed by running this code on an authenticated portal where the studies which a user has access to are a factor in the response from the database queries.

Reply:
I added two additional tests to verify behavior when the list is empty and populated